app/controllers/api/mixins/service_templates.rb

```diff
@@ -17,6 +19,14 @@ def order_service_template(id, data, scheduled_time = nil)
       def service_template_ident(st)
         "Service Template id:#{st.id} name:'#{st.name}'"
       end
+
+      def api_request_allowed?
+        standalone_api_request? ? Settings.product.allow_api_service_ordering : true
+      end
+
+      def standalone_api_request?
+        request.authorization.present? && request.authorization.try(:starts_with?, "basic")
```

Comment:
If we're doing the `.present?` check, why do we need the `.try`? Do we expect `request.authorization` not to respond to `starts_with?`?

Also, is the premise here valid @abellotti @jvlcek? Does external auth not work with the API? Or am I misunderstanding this check?

Reply:
Good question. I mentioned before when @bdunne suggested using the try that I pulled this line from https://github.com/rails/rails/blob/fc5dd0b85189811062c85520fd70de8389b55aeb/actionpack/lib/action_controller/metal/http_authentication.rb#L104.